adala/runtimes/_litellm.py

```diff
@@ -123,7 +120,40 @@ def _get_usage_dict(usage: Usage, model: str) -> Dict:
     return data
 
 
-class LiteLLMChatRuntime(Runtime):
+class InstructorClientMixin:
+    client: Any = Field(default=None, exclude=True)
+
+    def _from_litellm(self, **kwargs):
+        return instructor.from_litellm(litellm.completion, **kwargs)
+
+    def use_instructor_client(self) -> None:
+        if self.client is None:
+            kwargs = {}
+            if self.is_custom_openai_endpoint:
+                kwargs["mode"] = instructor.Mode.JSON
+            self.client = self._from_litellm(**kwargs)
+
+    @property
+    def is_custom_openai_endpoint(self) -> bool:
+        return self.model.startswith("openai/") and self.model_extra.get("base_url")
+
+    def __getstate__(self):
+        state = super().__getstate__()
+        state["__dict__"]["client"] = None
+        return state
+
+    def __setstate__(self, state):
+        super().__setstate__(state)
+        self.use_instructor_client()
```

Comment:
can you use `@field_serializer` or `@computed_field` like for the openai client in the runtimes instead of this deep magic?

Reply:
oh yeah, that's more concise, although with a little overhead due to the first call warm up, but maybe less error prone.